cmd/mirror-main.go

```diff
@@ -405,7 +405,27 @@ func (mj *mirrorJob) watchMirror(ctx context.Context, cancelMirror context.Cance
 				// Strip the prefixes in the event path. Happens in darwin OS only
 				eventPath = eventPath[strings.Index(eventPath, sourceURLFull):]
 			}
-
+			if runtime.GOOS == "windows" {
+				// Windows issue: Source URL is a URL-like address. For ex shared folder file://Users/username/foldername1/foldername2 or smb://workgroup;user:password@server/share/folder/file.txt
+				// But a rename or move or sometimes even write event sets eventPath as an absolute filepath.
+				// For object write, path conversion is necessary.
+				var eventRelPath, lastPathPrefix string
+				var lastPathPrefixPos int
+				sourceURLpathList := strings.Split(mj.sourceURL, slashSeperator)
+				lenSrcURLSlice := len(sourceURLpathList)
+				shdModifyEventPath := filepath.IsAbs(eventPath) && !filepath.IsAbs(mj.sourceURL) && lenSrcURLSlice > 1
+				if shdModifyEventPath {
+					lastPathPrefix = sourceURLpathList[lenSrcURLSlice-1]
+					lastPathPrefixPos = strings.Index(eventPath, lastPathPrefix)
+				}
+				canModifyEventPath := shdModifyEventPath && lastPathPrefix != "" && lastPathPrefixPos > 0
+				canModifyEventPath = canModifyEventPath && lastPathPrefixPos+len(lastPathPrefix) < len(eventPath)
+				if canModifyEventPath {
+					eventRelPath = eventPath[lastPathPrefixPos+len(lastPathPrefix):]
+					eventRelPath = strings.Replace(eventRelPath, "\\", "/", -1)
```

Comment:
filepath.ToSlash() is the helper you want @BigUstad

Reply:
Thanks.